File: DSA-for-Beginners/09_Hashing/01_Hash_Usage.py
def main():
    names = ["alice", "brad", "collin", "brad", "dylan", "kim"]

    countMap = {}
    for name in names:
        # If countMap does not contain name
        if name not in countMap:
            countMap[name] = 1
        else:
            countMap[name] += 1
    
    for key, value in countMap.items():
        print(f"{key}: {value}")

if __name__ == "__main__":
    main()

File: DSA-for-Beginners/09_Hashing/01_Hash_Usage.cpp
#include <iostream>
#include <vector>
#include <string>
#include <unordered_map>

using std::vector;
using std::string;
using std::unordered_map;
using std::cout;
using std::endl;

int main() {
    vector<string> names = {"alice", "brad", "collin", "brad", "dylan", "kim"};
    unordered_map<string, int> countMap;

    for (string& name : names) {
        if (countMap.count(name) == 0) {
            countMap[name] = 1;
        } else {
            countMap[name]++;
        }
    }

    for (auto& pair : countMap) {
        cout << pair.first << ": " << pair.second << endl;
    }
    
    return 0;
}